src/catalog/catalog_content.cpp

```diff
@@ -232,5 +235,27 @@ void CatalogContent::registerBuiltInFunctions() {
     builtInTableFunctions = std::make_unique<function::BuiltInTableFunctions>();
 }
 
+std::vector<common::table_id_t> CatalogContent::getAllTableIDsByType(TableType tableType) const {
+    std::vector<common::table_id_t> tableIDs;
+    for (auto& tableSchema : tableSchemas) {
+        if (tableSchema.second->getTableType() == tableType) {
+            tableIDs.push_back(tableSchema.first);
+        }
+    }
+    return tableIDs;
+}
+
+void CatalogContent::validateTableID(common::table_id_t tableID) const {
```

Comment:
The two validations doesn't have test coverage. Consider use assert if they are not trigger-able